hera_pspec/pspecdata.py

```diff
@@ -560,6 +586,34 @@ def cross_covar_model(self, key1, key2, model='empirical', conj_1=False, conj_2=
             covar = utils.cov(self.x(key1), self.w(key1),
                               self.x(key2), self.w(key2),
                               conj_1=conj_1, conj_2=conj_2)
+        if model == 'time_average':
+            x1 = self.x(key1)
+            #(Nspw_freqs, Ntimes)
+            x2 = self.x(key2)
+            #(Nspw_freqs, Ntimes)
+            x1 = x1.T
+            #(Ntimes, Nspw_freqs)
+            x2 = x2.T
+            #(Ntimes, Nspw_freqs)
+            if conj_1:
+                x1 = x1.conj()
+            if conj_2:
+                x2 = x2.conj()
+            x1x2 = np.zeros((x1.shape[0], x1.shape[1], x1.shape[1]), dtype=np.complex128)
+            #(Ntimes, Nspw_freqs, Nspw_freqs)
+            for time in range(x1.shape[0]):
+                x1x2[time, :, :] = np.einsum('i,j', x1[time, :], x2[time, :])
+            covar = np.average(x1x2, axis=0)
+            #(Nspw_freqs, Nspw_freqs)
+            if subtracted == True:
+                x1_average = np.average(x1, axis=0)
```

Comment:
I think my comment above also applies here: you might consider weighting the data with `self.w(key)` before taking the time average, and then performing the time-average as described above.